src/core/proj/qgsprojutils.cpp

```diff
@@ -259,6 +260,14 @@ QgsProjUtils::proj_pj_unique_ptr QgsProjUtils::crsToDatumEnsemble( const PJ *crs
 #endif
 }
 
+QgsProjUtils::proj_pj_unique_ptr QgsProjUtils::createCompoundCrs( PJ *horizontalCrs, PJ *verticalCrs )
```

Comment:
I was going to suggest to accept a 'const PJ*' instead of a 'PJ*', but I realized the issue was in the PROJ API itself. This will be repaired by https://github.com/OSGeo/PROJ/pull/4074. So fundamentally PROJ will not mess with the passed input CRS, and you could safely const_cast<PJ*> to make it work with current proj.h API. Anyway that's nitpicking...
PR LGTM

Reply:
I did wonder about that! const_cast added...